src/Symfony/Component/Yaml/Yaml.php

```diff
@@ -41,21 +41,28 @@ class Yaml
      * @param string $input                  Path to a YAML file or a string containing YAML
      * @param bool   $exceptionOnInvalidType True if an exception must be thrown on invalid types false otherwise
      * @param bool   $objectSupport          True if object support is enabled, false otherwise
+     * @param bool   $objectForMap           True if maps should return a stdClass instead of array()
+     * @param bool   $timestampAsDateTime    True if timestamps must be parsed as DateTime objects rather than Unix timestamps (integers)
      *
      * @return array The YAML converted to a PHP array
      *
      * @throws ParseException If the YAML is not valid
      *
-     * @deprecated The ability to pass file names to the Yaml::parse method is deprecated since version 2.2 and will be removed in 3.0. Pass the YAML contents of the file instead.
+     * @deprecated The ability to pass file names to the Yaml::parse method is deprecated since version 2.7 and will be removed in 3.0. Pass the YAML contents of the file instead.
+     * @deprecated The ability to pass $timestampAsDateTime = false to the Yaml::parse method is deprecated since version 2.8. The argument will be removed in 3.0. Pass true instead.
      *
      * @api
      */
-    public static function parse($input, $exceptionOnInvalidType = false, $objectSupport = false)
+    public static function parse($input, $exceptionOnInvalidType = false, $objectSupport = false, $objectForMap = false, $timestampAsDateTime = false)
     {
+        if (!$timestampAsDateTime) {
```

Comment:
What do you think:
```php
if (func_num_args() > 4) {
    // trigger_error(...);
}
```
this doesn't work if we add another param to the function, but it's something